joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/typesolvers/noncaching/JdkJarTypeSolver.scala

```diff
@@ -84,26 +90,28 @@ class JdkJarTypeSolver private (jdkPath: String) extends TypeSolver {
     }
   }
 
-  private def addJarToClassPool(path: String): Unit = {
-    path match {
-      case jarPath if jarPath.endsWith(".jar") =>
-        classPool.appendClassPath(jarPath)
+  private def addPathToClassPool(archivePath: String): Try[ClassPath] = {
+    if (archivePath.isJarPath) Try(classPool.appendClassPath(archivePath))
+    else if (archivePath.isJmodPath)
+      val classPath = new JmodClassPath(archivePath)
+      Try(classPool.appendClassPath(classPath))
+    else Failure(new IllegalArgumentException("$archivePath is not a path to a jar/jmod"))
+  }
 
-      case jmodPath =>
-        classPool.appendClassPath(new JdkArchiveClassPath(jmodPath))
-    }
+  def withJars(archivePaths: Seq[String]): JdkJarTypeSolver = {
+    addArchives(archivePaths)
+    this
   }
 
-  private def withJars(jarPaths: Seq[String]): JdkJarTypeSolver = {
-    jarPaths.foreach { jarPath =>
-      Try(classPool.appendClassPath(new JdkArchiveClassPath(jarPath))) match {
-        case Success(_) => registerKnownClassesForJar(jarPath)
+  def addArchives(archivePaths: Seq[String]): Unit = {
+    archivePaths.foreach { archivePath =>
+      Try(addPathToClassPool(archivePath)) match {
```

Comment:
This `Try()` wrapping seems unnecessary as `addPathToClassPool` already returns a `Try`. If the wrapping is only to catch possible exceptions coming from `new JmodClassPath(...` in line 96, than i suggest you do the transition from exception to `Try` there.

Reply:
Good point. Updating the PR